packages/apps/human-app/frontend/src/api/fetcher.ts

```diff
@@ -71,6 +69,22 @@ export function createFetcher(defaultFetcherConfig?: {
   options?: RequestInit | (() => RequestInit);
   baseUrl: FetcherUrl | (() => FetcherUrl);
 }) {
+  let refreshPromise: Promise<SignInSuccessResponse | undefined> | null = null;
+
+  async function refreshToken(): Promise<
```

Comment:
As per [this comment](https://github.com/humanprotocol/human-protocol/pull/2824/files#r1853643880), now we can move it out of `createFetcher` and below it will still
```
return fetcher
```

Reply:
Just to clarify: we want to actually separate `fetcher` from `refresh` completely? Go back from `apiClient.fetcher` to `apiClient` and use explicit `refreshToken` rather than `apiClient.refreshToken`?

Reply:
Sure, sorry for that

Reply:
Thanks for jumping into with clarifications, let me share my though process :)

From naming `createFetcher` - I would expected that such "factory function" creates and returns just a "fetcher" instance that I would later user to make some requests, this is why I initially though about moving it out (+the fact that `refreshToken` doesn't use anything from `createFetcher` options at all, so why it would be there)

> Go back from apiClient.fetcher to apiClient and use explicit refreshToken rather than apiClient.refreshToken?

See, here you have different naming - `apiClient`, and when having such naming it actually makes sense to have: `apiClient.fetch()` and `apiClient.refreshToken()`

So no need to move it out, we can just remove that confusion by:
- using `baseUrl` in `fetchTokenRefresh()` instead of `env.VITE_API_URL`, since we create `apiClient` with specific config
- rename `createFetcher` to `createApiClient`, because it's not just generic "fetcher", it's highly coupled to the API it uses
- return `{ fetch, refreshToken }`

This way we will be explicit on what we have, wdyt?

Reply:
@mpblocky Thanks for jumping into with clarifications, let me share my though process :)

From naming `createFetcher` - I would expected that such "factory function" creates and returns just a "fetcher" instance that I would later user to make some requests, this is why I initially though about moving it out (+the fact that `refreshToken` doesn't use anything from `createFetcher` options at all, so why it would be there)

> Go back from apiClient.fetcher to apiClient and use explicit refreshToken rather than apiClient.refreshToken?

See, here you have different naming - `apiClient`, and when having such naming it actually makes sense to have: `apiClient.fetch()` and `apiClient.refreshToken()`

So no need to move it out, we can just remove that confusion by:
- using `baseUrl` in `fetchTokenRefresh()` instead of `env.VITE_API_URL`, since we create `apiClient` with specific config
- rename `createFetcher` to `createApiClient`, because it's not just generic "fetcher", it's highly coupled to the API it uses
- return `{ fetch, refreshToken }`

This way we will be explicit on what we have, wdyt?

Reply:
Both approaches make sense actually - it boils down to if we want to have explicit usage of `env` in fetchRefreshToken or have a singular place where we take it from

Reply:
True, so feel free to pick any of them, just not the mix 😅

Reply:
lets stay with the current state - there would be additional changes needed and with the current code I'm not sure how much changes would be needed :sweat_smile: